code/go/0chain.net/chaincore/block/block_eventdb.go

```diff
@@ -0,0 +1,54 @@
+package block
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"0chain.net/smartcontract/dbs/event"
+)
+
+func blockToBlockEvent(block *Block) *event.Block {
+	return &event.Block{
+		Hash:                  block.Hash,
+		Version:               block.Version,
+		CreationDate:          int64(block.CreationDate.Duration()),
+		Round:                 block.Round,
+		MinerID:               block.MinerID,
+		RoundRandomSeed:       block.RoundRandomSeed,
+		MerkleTreeRoot:        block.GetMerkleTree().GetRoot(),
+		StateHash:             string(block.ClientStateHash),
+		ReceiptMerkleTreeRoot: block.GetReceiptsMerkleTree().GetRoot(),
+		NumTxns:               len(block.Txns),
+		MagicBlockHash:        block.MagicBlock.Hash,
+		PrevHash:              block.PrevHash,
+		Signature:             block.Signature,
+		ChainId:               block.ChainID,
+		ChainWeight:           fmt.Sprintf("%f", block.ChainWeight),
+		RunningTxnCount:       fmt.Sprintf("%d", block.RunningTxnCount),
+		RoundTimeoutCount:     block.RoundTimeoutCount,
+		CreatedAt:             block.CreationDateField.ToTime(),
+	}
+}
+
+func emitBlockEvent(block *Block) error {
+	if len(block.Txns) == 0 {
+		return fmt.Errorf("no transaction for the block")
+	}
+
+	data, err := json.Marshal(blockToBlockEvent(block))
+	if err != nil {
+		return fmt.Errorf("error marshalling block: %v", err)
+	}
+
+	t := block.Txns[len(block.Txns)-1]
+	block.Events = append(block.Events, event.Event{
+		BlockNumber: block.Round,
+		TxHash:      t.Hash,
+		Type:        int(event.TypeStats),
+		Tag:         int(event.TagAddBlock),
+		Index:       t.Hash,
```

Comment:
I am passing the transaction hash of last transaction. Is it okay or I am missing anything ?

Reply:
Use the block's hash as an index. b.Hash. 

There is no meaning for the `Event.TxHash` field, so set to `""`, we can revisit if that breaks anything. However, it seems a reasonable way to indicate an event that is not linked to a particular transaction.